cpp/jxrdecode.cpp

```diff
@@ -66,22 +65,28 @@ void stream_data() {
   print_bytes(bytes);
   std::cerr << std::endl;
 
-  ImageDecoder decoder = codecFactory.decoderFromBytes(bytes);
+  ImageDecoder decoder;
+  codecFactory.decoderFromBytes(decoder, bytes);
   std::cerr << "Opened decoder with " << bytes.size() << " bytes" << std::endl;
 
+  unsigned int frameSize =
+    decoder.getWidth() * decoder.getHeight() * decoder.getBytesPerPixel();
+  unsigned char *image_buffer = new unsigned char[frameSize];
```

Comment:
Instead of using a `unsigned char[]`, you could use `std::vector<unsigned char>` and then pass the return of `vector::data()` for the decode (vector's underlying storage is guaranteed to be contiguous).